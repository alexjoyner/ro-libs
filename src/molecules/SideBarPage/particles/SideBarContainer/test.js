import React from 'react';
import { shallow } from 'enzyme';
import { SideBarContainer } from './';

describe('SideBarContainer component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<SideBarContainer />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('should default to correct position', () => {
    expect(component).toHaveStyleRule('left', '-250px');
  });
  it('should accept shown prop and change style', () => {
    const shownTest = shallow(<SideBarContainer shown />);
    expect(shownTest).toHaveStyleRule('left', '0px');
  });
});
