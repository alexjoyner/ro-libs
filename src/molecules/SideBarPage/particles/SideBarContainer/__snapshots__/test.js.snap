// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarContainer component Should match snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: 1;
  top: 56px;
  -webkit-transition: 100ms linear;
  transition: 100ms linear;
  height: 100vh;
  width: 300px;
  background-color: #2d3436;
  color: white;
  font-size: 20px;
  left: -250px;
}

.c0 ul {
  list-style: none;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  position: fixed;
  z-index: ",
            "1",
            ";
  top: 56px;
  transition: 100ms linear;
  height: 100vh;
  width: 300px;
  background-color: ",
            [Function],
            ";
  color: white;
  font-size: 20px;
  ul{
      list-style: none;
  }
  ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;
